data/crac-file/crac-io-json/src/test/java/com/farao_community/farao/data/crac_io_json/JsonImportExportTest.java

```diff
@@ -0,0 +1,198 @@
+        threshold1.setMaxValue(999);
+        VoltageThreshold threshold2 = new VoltageThreshold(280, 300);
+        threshold2.setMinValue(275);
+        threshold2.setMaxValue(305);
+
+        // CNECs
+        SimpleCnec cnec1 = new SimpleCnec("idCnec", "Cnec", null, threshold1, stateCurative);
+        cnec1.setCriticalNetworkElement(monitoredElement);
+        SimpleCnec cnec2 = new SimpleCnec("idCnec2", "Cnec 2", monitoredElement, null, null);
+        cnec2.setState(stateBasecase);
+        cnec2.setThreshold(threshold2);
+
+        // Usage rules
+        FreeToUse freeToUse = new FreeToUse(null, null);
+        freeToUse.setUsageMethod(UsageMethod.AVAILABLE);
+        freeToUse.setState(stateBasecase);
+        OnContingency onContingency = new OnContingency(UsageMethod.FORCED, stateCurative, null);
+        onContingency.setContingency(contingency);
+        OnConstraint onConstraint = new OnConstraint(UsageMethod.FORCED, stateCurative, null);
+        onConstraint.setCnec(cnec1);
+
+        // NetworkAction
+        ComplexNetworkAction networkAction1 = new ComplexNetworkAction("id1", "name1", new ArrayList<>(Arrays.asList(freeToUse)), new ArrayList<>(Arrays.asList(hvdcSetpoint)));
+        networkAction1.addApplicableNetworkAction(topology2);
+        ComplexNetworkAction networkAction2 = new ComplexNetworkAction("id2", "name2", new ArrayList<>(Arrays.asList(freeToUse)), new ArrayList<>(Arrays.asList(pstSetpoint)));
+
+        // RangeAction
+        ComplexRangeAction rangeAction1 = new ComplexRangeAction("idRangeAction", "myRangeAction", null, null, null);
+        List<Range> ranges = new ArrayList<>(Arrays.asList(absoluteFixedRange, relativeDynamicRange));
+        rangeAction1.setRanges(ranges);
+        rangeAction1.addRange(relativeFixedRange);
+        List<ApplicableRangeAction> elementaryRangeActions = new ArrayList<>(Arrays.asList(pstRange1));
+        rangeAction1.setApplicableRangeActions(elementaryRangeActions);
+        rangeAction1.addApplicableRangeAction(hvdcRange1);
+        List<AbstractUsageRule> usageRules = new ArrayList<>(Arrays.asList(freeToUse, onConstraint));
+        rangeAction1.setUsageRules(usageRules);
+        rangeAction1.addUsageRule(onContingency);
+
+        ComplexRangeAction rangeAction2 = new ComplexRangeAction("idRangeAction2", "myRangeAction2", usageRules, ranges, new ArrayList<>(Arrays.asList(pstRange1)));
+
+        List<Cnec> cnecs = new ArrayList<>();
+        cnecs.add(cnec1);
+
+        SimpleCrac crac = new SimpleCrac("idCrac", "name", new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
+
+        crac.setCnecs(cnecs);
+        crac.addCnec(cnec2);
+        crac.setRangeActions(new ArrayList<>(Arrays.asList(rangeAction1)));
+        crac.addRangeRemedialAction(rangeAction2);
+
+        return crac;
+    }
+
+    @Test
+    public void testExportCrac() throws IOException {
+
+        Crac crac = create();
+
+        OutputStream os = new FileOutputStream("file.json");
```

Comment:
It might be better to not generate a new file.json during the test?
Could use ByteArrayOutputStream() and compare the string?

Reply:
This is more related to Viktor's PR and I think it will be changed in this PR instead